libcudacxx/docs/extended_api/synchronization_primitives/barrier/memcpy_async_tx.md

```diff
@@ -0,0 +1,87 @@
+---
+grand_parent: Extended API
+parent: Barriers
+---
+
+# `cuda::device::memcpy_async_tx`
+
+Defined in header `<cuda/barrier>`:
+
+```cuda
+template <typename T, size_t Alignment>
+inline __device__
+async_contract_fulfillment
+cuda::device::memcpy_async_tx(
+  T* dest,
+  const T* src,
+  cuda::aligned_size_t<Alignment> size,
+  cuda::barrier<cuda::thread_scope_block>& bar);
```

Comment:
barrier is a template, second parameter missing

Reply:
This function requires an mbarrier, we do not use mbarrier when the completion function is not the default one. This is fine as is.